src/components/QuickOpenModal.js

```diff
@@ -111,18 +111,10 @@ export class QuickOpenModal extends Component<Props, State> {
   };
 
   searchSources = (query: string) => {
-    if (query == "") {
-      const results = this.props.sources;
-      return this.setState({ results });
-    }
-    if (this.isGotoSourceQuery()) {
-      const [baseQuery] = query.split(":");
-      const results = filter(this.props.sources, baseQuery);
-      this.setState({ results });
-    } else {
-      const results = filter(this.props.sources, query);
-      this.setState({ results });
-    }
+    const { sources } = this.props;
+    const results =
+      query == "" ? sources : filter(sources, query.split(":")[0]);
```

Comment:
lets move this to its own function `query.split(":")[0]`

`filter(sources, dropGoto(query))`

i dont like the term drop but it is nice to think about the language we use